packages/access/lib/mail.js

```diff
@@ -135,30 +155,47 @@ const getGlobalMergeVars = async (
   grantee, recipient, campaign, grant, t, pgdb
 ) => {
   const safeGrantee = transformUser(grantee)
+  const safeRecipient = !!recipient && transformUser(recipient)
   const recipientCampaigns =
     !!recipient && await campaignsLib.findForGrantee(recipient, { pgdb })
   const recipientHasMemberships =
     !!recipient && await membershipsLib.hasUserActiveMembership(recipient, pgdb)
 
+  const email = recipient ? recipient.email : grant.email
+
   return [
-    // Grant
+    // Grant,
+    { name: 'GRANT_CREATED',
+      content: dateFormat(grant.createdAt)
+    },
+    { name: 'GRANT_BEGIN_BEFORE',
+      content: dateFormat(grant.beginBefore)
+    },
     { name: 'GRANT_BEGIN',
-      content: dateFormat(grant.beginAt)
+      content: grant.beginAt && dateFormat(grant.beginAt)
     },
     { name: 'GRANT_END',
-      content: dateFormat(grant.endAt)
+      content: grant.endAt && dateFormat(grant.endAt)
+    },
+    { name: 'GRANT_VOUCHER_CODE',
+      content: grant.voucherCode
     },
 
     // Grantee
     { name: 'GRANTEE_NAME',
-      content: safeGrantee.name ||
-        t('api/noname')
+      content: safeGrantee.name || t('api/noname')
+    },
+    { name: 'GRANTEE_MESSAGE',
+      content: !!grant.message && grant.message.trim().replace(/\n/g, '<br />')
     },
 
     // Recipient
     { name: 'RECIPIENT_EMAIL',
       content: grant.email
     },
+    { name: 'RECIPIENT_NAME',
+      content: safeRecipient.name || t('api/noname')
```

Comment:
Can't reproduce `null`. What made you write this comment?

`safeRecipient` is either
- an object, as `transformUser` will turn all truthy values into an object
- `false`, in case `recipient` turns out falsy on line 158

```
> transformUser('peter parker')
{ id: undefined,
  username: undefined,
  firstName: undefined,
  lastName: undefined,
  name: '',
  hasPublicProfile: undefined,
  roles: [],
  email: undefined,
  _raw: 'peter parker' }
```

```
> const recipient = null
> const safeRecipient = !!recipient
> console.log(safeRecipient.name || 'fallback')
fallback
```